tests/test_code_gen.py

```diff
@@ -271,11 +271,11 @@ def test_pow(self):
 
     def test_comprehension(self):
         source = """
-            ((x,y) for x,y in zip(a,b))
+            ((x,y) for x,y in zip(a,b) if a)
```

Comment:
Really, it was just to also test the case where an `if` is present, since it was entirely untested before (thought admittedly not too much can go wrong there...).